langchain/src/schema/runnable/passthrough.ts

```diff
@@ -1,6 +1,85 @@
-import { Runnable } from "./base.js";
+import { IterableReadableStream } from "../../util/stream.js";
+import { Runnable, RunnableMap } from "./base.js";
 import type { RunnableConfig } from "./config.js";
 
+/**
+ * A runnable that assigns key-value pairs to inputs of type `Record<string, unknown>`.
+ */
+export class RunnableAssign<
+  RunInput = unknown,
+  RunOutput = unknown,
+  CallOptions extends RunnableConfig = RunnableConfig
+> extends Runnable<RunInput, RunOutput> {
+  lc_namespace = ["langchain", "schema", "runnable"];
+
+  mapper: RunnableMap<RunInput>;
+
+  constructor(mapper: RunnableMap<RunInput>) {
+    super();
+    this.mapper = mapper;
+  }
+
+  async invoke(input: RunInput, options?: Partial<CallOptions>) {
+    const mapperResult = await this.mapper.invoke(input, options);
+
+    return {
+      ...input,
+      ...mapperResult,
+    } as RunOutput;
+  }
+
+  async *transform(
```

Comment:
Will want a test case for this, a lot of subtleties here

Reply:
Also, is this implemented generally for runnable maps yet in JS?

Reply:
Yeah doesn't look like maps support stream/transform chunks yet - would be good to add at some point but separately.

Reply:
Removed entirely, issue #2978 is tracking implementing it in runnable map & assign